lib/ClangImporter/ImportDecl.cpp

```diff
@@ -5636,6 +5660,77 @@ Decl *SwiftDeclConverter::importGlobalAsInitializer(
   return result;
 }
 
+Decl *SwiftDeclConverter::importCXXMethodDecl(
+    const clang::CXXMethodDecl *decl, DeclName name, DeclContext *dc,
+    Optional<ImportedName> correctSwiftName) {
+  auto isStatic = decl->isStatic();
+  auto loc = Impl.importSourceLoc(decl->getLocation());
+  if (dc->getSelfProtocolDecl() && isStatic) {
+    Impl.SwiftContext.Diags.diagnose(loc, diag::swift_name_protocol_static,
+                                     /*isInit=*/false);
+    Impl.SwiftContext.Diags.diagnose(loc, diag::note_while_importing,
+                                     decl->getName());
+    return nullptr;
+  }
+
+  if (!decl->hasPrototype()) {
+    Impl.SwiftContext.Diags.diagnose(loc, diag::swift_name_no_prototype);
+    Impl.SwiftContext.Diags.diagnose(loc, diag::note_while_importing,
+                                     decl->getName());
+    return nullptr;
+  }
+
+  bool allowNSUIntegerAsInt =
+      Impl.shouldAllowNSUIntegerAsInt(isInSystemModule(dc), decl);
+
+  auto &C = Impl.SwiftContext;
+  auto *bodyParams = Impl.importFunctionParameterList(
+      dc, decl, {decl->param_begin(), decl->param_end()}, decl->isVariadic(),
+      allowNSUIntegerAsInt, name.getArgumentNames());
+
+  auto importedType =
+      Impl.importFunctionReturnType(dc, decl, allowNSUIntegerAsInt);
+  Type swiftResultTy = importedType.getType();
+
+  auto nameLoc = Impl.importSourceLoc(decl->getLocation());
+  auto result = createFuncOrAccessor(C, loc, None, name, nameLoc, bodyParams,
+                                     swiftResultTy,
+                                     /*throws*/ false, dc, decl);
+
+  result->setGenericEnvironment(dc->getGenericEnvironmentOfContext());
+
+  result->setAccess(AccessLevel::Public);
```

Comment:
Is there any logic which causes us to skip importing non-`public` methods?

Reply:
I added something to import this as private, but to the best I could figure out this skipping non-public  anything seems woefully incomplete. (non-public base classes, non-public fields, etc). I'll have to address that more completely in a followup PR.

Reply:
Well, non-public fields should maybe still be imported, just as `private`; but there's no reason to import non-public methods at all.

Reply:
Alright, sounds good. Perhaps with good copy/move-constructor support, private fields could be skipped as well.